nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AnnotationServiceImpl.java

```diff
@@ -103,7 +103,13 @@ public Annotation getAnnotation(CoreSession session, String annotationId) {
 
     @Override
     public void updateAnnotation(CoreSession session, String annotationId, Annotation annotation) {
-        Framework.getService(CommentManager.class).updateComment(session, annotationId, annotation);
+        IdRef annotationRef = new IdRef(annotationId);
+        if (!session.exists(annotationRef)) {
+            throw new IllegalArgumentException("The annotation " + annotationId + " does not exist.");
+        }
+        DocumentModel annotationModel = session.getDocument(annotationRef);
+        Comments.externalAnnotationToDocumentModel().accept(annotation, annotationModel);
```

Comment:
Sorry this is about code previously committed, but I have to ask: is there a reason why all these methods in `Comments` return functions that you then have to call? Do we pass them around as values somewhere? If not, it's simpler to just make the methods do the work directly instead of getting always getting a function and calling `accept` on it...

Reply:
We thought it was have been useful, but we don't have any real use case in the end. I will change it to have simpler methods.